hack/generate/samples/helm/memcached.go

```diff
@@ -70,49 +70,53 @@ func (mh *MemcachedHelm) Run() {
 	// required to make Helm project generation idempotent because contributors and CI environments
 	// can all have slightly different environments that can affect the content of the generated
 	// role and cause sanity testing to fail.
-	os.Setenv("KUBECONFIG", "broken_so_we_generate_static_default_rules")
+	if strings.Contains(mh.ctx.Dir, "testdata") {
+		// Set the env var only when it is running to gen the sample
+		// For the e2e test the following should not be set
+		os.Setenv("KUBECONFIG", "broken_so_we_generate_static_default_rules")
+	}
 
 	log.Infof("creating the project")
 	err = mh.ctx.Init(
 		"--plugins", "helm",
 		"--domain", mh.ctx.Domain)
-	pkg.CheckError("creating the project", err)
+	samples.CheckError("creating the project", err)
 
 	log.Infof("handling work path to get helm chart mock data")
 	projectPath := strings.Split(current, "operator-sdk/")[0]
 	projectPath = strings.Replace(projectPath, "operator-sdk", "", 1)
-	helmChartPath := filepath.Join(projectPath, "operator-sdk/hack/generate/samples/internal/helm/testdata/memcached-0.0.1.tgz")
+	helmChartPath := filepath.Join(projectPath, "operator-sdk/hack/generate/samples/helm/testdata/memcached-0.0.1.tgz")
```

Comment:
Somewhat tangential to this PR, but why are we going to all this trouble to find the project root? I would expect a hardcoded relative path here. If I happen to clone the repo in a directory path like `/home/joe/operator-sdk/operator-sdk`, this will fail.

Reply:
I agree that it is not the best solution. The problem is that in the tests the current path is not the same. I will try to improve that.

Reply:
I did a few improvements but it might be solved soon in the cleanups to perform changes as you suggested.